doc/faq.rst

```diff
@@ -707,4 +707,139 @@ Alpine 3.9 image pulled from Docker hub::
+realistic example.
+
+SSH stops :code:`ch-image` build with interactive queries
+---------------------------------------------------------
+
+This often occurs during an SSH-based Git clone. For example:
+
+.. code-block:: docker
+
+  FROM alpine:latest
+  RUN apk add git openssh
+  RUN git clone git@github.com:hpc/charliecloud.git
+
+.. code-block:: console
+
+  $ ch-image build -t foo -f ./Dockerfile .
+  [...]
+  3 RUN ['/bin/sh', '-c', 'git clone git@github.com:hpc/charliecloud.git']
+  Cloning into 'charliecloud'...
+  The authenticity of host 'github.com (140.82.113.3)' can't be established.
+  RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
+  Are you sure you want to continue connecting (yes/no/[fingerprint])?
+
+At this point, the build stops while SSH waits for input.
+
+This happens even if you have :code:`github.com` in your
+:code:`~/.ssh/known_hosts`. This file is not available to the build because
+:code:`ch-image` runs :code:`ch-run` with :code:`--no-home`, so :code:`RUN`
+instructions can't see anything in your home directory.
+
+Solutions include:
+
+  1. Change to anonymous HTTPS clone, if available. Most public repositories
+     will support this. For example:
+
+     .. code-block:: docker
+
+       FROM alpine:latest
+       RUN apk add git
+       RUN git clone https://github.com/hpc/charliecloud.git
+
+  2. Approve the connection interactively by typing :code:`yes`. Note this
+     will record details of the connection within the image, including IP
+     address and the fingerprint. The build also remains interactive.
+
+  3. Edit the image's system `SSH config
+     <https://man.openbsd.org/ssh_config>`_ to turn off host key checking.
+     Note this can be rather hairy, because the SSH config language is quite
+     flexible and the first instance of a directive is the one used. However,
+     often the changes can be simply appended:
+
+     .. code-block:: docker
+
+       FROM alpine:latest
+       RUN apk add git openssh
+       RUN printf 'StrictHostKeyChecking=no\nUserKnownHostsFile=/dev/null\n' \
+           >> /etc/ssh/ssh_config
+       RUN git clone git@github.com:hpc/charliecloud.git
+
+     Check your institutional policy on whether this is permissible, though
+     it's worth noting that users `almost never
+     <https://www.usenix.org/system/files/login/articles/105484-Gutmann.pdf>`_
+     verify the host fingerprints anyway.
+
+     This will not record details of the connection in the image.
+
+  4. Turn off host key checking on the SSH command line. (See caveats in the
+     previous item.) The wrapping tool should provide a way to configure this
+     command line. For example, for Git:
+
+     .. code-block:: docker
+
+       FROM alpine:latest
+       RUN apk add git openssh
+       ARG GIT_SSH="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+       RUN git clone https://github.com/hpc/charliecloud.git
```

Comment:
Just noticed this typo; sorry I missed it before!  It's a `GIT_SSH` example but is cloning via `https`; should read `RUN git clone git@github.com:hpc/charliecloud.git` instead.